paper/src/main/java/com/denizenscript/denizen/paper/events/BlockPreDispenseScriptEvent.java

```diff
@@ -0,0 +1,74 @@
+package com.denizenscript.denizen.paper.events;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizen.objects.LocationTag;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import io.papermc.paper.event.block.BlockPreDispenseEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class BlockPreDispenseScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // <block> tries to dispense <item>
+    //
+    // @Group Block
+    //
+    // @Location true
+    //
+    // @Cancellable true
+    //
+    // @Triggers before a block dispenses an item.
+    //
+    // @description
+    // This event fires before the dispenser finishes processing its drop, allowing access to the slot trying to be dispensed and full cancellation of the dispenser's sounds.
```

Comment:
I don't think this is a thing? should be under `Triggers` afaik ([see](https://github.com/DenizenScript/Denizen/blob/9afda6bf726e841356b4fc0c12ce1d0c8c2f7842/plugin/src/main/java/com/denizenscript/denizen/events/entity/EntityDeathScriptEvent.java#L39-L40))